src/backend/distributed/operations/stage_protocol.c

```diff
@@ -292,8 +292,16 @@ master_append_table_to_shard(PG_FUNCTION_ARGS)
 	ShardPlacement *shardPlacement = NULL;
 	foreach_ptr(shardPlacement, shardPlacementList)
 	{
-		MultiConnection *connection = GetPlacementConnection(FOR_DML, shardPlacement,
-															 NULL);
+		int connectionFlags = FOR_DML;
+		MultiConnection *connection =
+			GetPlacementConnection(connectionFlags, shardPlacement, NULL);
+
+		/*
+		 * This code-path doesn't support optional connections, so we don't expect
+		 * NULL connections.
+		 */
+		Assert(connection != NULL);
```

Comment:
Could we introduce a GetOptionalPlacementConnection to not have to worry about calling GetPlacementConnection? (the latter still guarantees not-NULL result)

Reply:
I think it would be nicer to keep the commonly used API (GetPlacementConnection) simpler. It should assert that OPTIONAL_CONNECTION is not set.

Reply:
Makes sense, will do that

Reply:
As talked, I'll do it with a follow-up PR not to complicate this PR